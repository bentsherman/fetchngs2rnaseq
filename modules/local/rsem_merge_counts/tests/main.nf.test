nextflow_process {

    name "Test Process RSEM_MERGE_COUNTS"
    script "../main.nf"
    process "RSEM_MERGE_COUNTS"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.of([
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/genes/RAP1_IAA_30M_REP1.genes.results", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/genes/RAP1_UNINDUCED_REP1.genes.results", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/genes/RAP1_UNINDUCED_REP2.genes.results", checkIfExists: true)
                ]).collect()
                input[1] = Channel.of([
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/isoforms/RAP1_IAA_30M_REP1.isoforms.results", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/isoforms/RAP1_UNINDUCED_REP1.isoforms.results", checkIfExists: true),
                    file(params.pipelines_testdata_base_path + "rsem_merge_counts/isoforms/RAP1_UNINDUCED_REP2.isoforms.results", checkIfExists: true)
                ]).collect()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
