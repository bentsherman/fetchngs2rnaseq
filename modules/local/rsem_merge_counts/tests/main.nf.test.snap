{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "rsem.merged.gene_counts.tsv:md5,7d1da94077dc2f90cfb2c793ca5b7446"
                ],
                "1": [
                    "rsem.merged.gene_tpm.tsv:md5,39bad606eb012456bba1d995fe0feb5f"
                ],
                "2": [
                    "rsem.merged.transcript_counts.tsv:md5,e40bba0aafc5904361513b3513c217ad"
                ],
                "3": [
                    "rsem.merged.transcript_tpm.tsv:md5,abbaac45f9938716c58d604299ea284e"
                ],
                "4": [
                    "versions.yml:md5,48ca3e12c91829af8019462b3f6aa29c"
                ],
                "counts_gene": [
                    "rsem.merged.gene_counts.tsv:md5,7d1da94077dc2f90cfb2c793ca5b7446"
                ],
                "counts_transcript": [
                    "rsem.merged.transcript_counts.tsv:md5,e40bba0aafc5904361513b3513c217ad"
                ],
                "tpm_gene": [
                    "rsem.merged.gene_tpm.tsv:md5,39bad606eb012456bba1d995fe0feb5f"
                ],
                "tpm_transcript": [
                    "rsem.merged.transcript_tpm.tsv:md5,abbaac45f9938716c58d604299ea284e"
                ],
                "versions": [
                    "versions.yml:md5,48ca3e12c91829af8019462b3f6aa29c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-09T17:13:37.377348"
    }
}